src/v/config/configuration.cc

```diff
@@ -1160,6 +1160,12 @@ configuration::configuration()
       "Enable re-uploading data for compacted topics",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       true)
+  , cloud_storage_recovery_temporary_retention_ms(
+      *this,
+      "cloud_storage_recovery_temporary_retention_ms",
+      "Retention in milliseconds for topics created during automated recovery",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      60min)
```

Comment:
The situation where this process runs is that we have a new cluster where there are no topics, and we have some data in a bucket which we need to recovery.

The bucket is scanned and the topics are auto-created during the recovery process. Then the content for these new topics is download by the partition recovery manager, which uses this retention period to decide how much data it should download. This short retention is used to drive the download process, so that the recovery manager downloads one hour's worth of data and exits. 

If we do not set this value the default of one week will be used which is a large amount of data to download.

Once the download is complete, the topic is reset to the original retention setting read from the topic manifest, which could be much higher.

So this setting is not used for GC (although technically it could kick in, if a producer was started during the recovery and the recovery ran for over one hour). 

This retention period is kind of arbitrary though. If given this description you think that it should be higher, we can discuss and increase this.

Reply:
This is _local_ retention, right?  i.e. it's a temporary override for `retention.local.target.ms`?

Reply:
I used this value for both `retention_local_target_ms` and `retention_duration`. Should it only be used for the `retention_local_target_ms` configuration?

The change is https://github.com/redpanda-data/redpanda/blob/b0143dd16523c9fdce258250531c53c3474e0881/src/v/cloud_storage/topic_recovery_service.cc#L336..L348 and https://github.com/redpanda-data/redpanda/blob/b0143dd16523c9fdce258250531c53c3474e0881/src/v/cloud_storage/topic_recovery_service.cc#L324-L329